velox/dwio/common/tests/utils/BatchMaker.cpp

```diff
@@ -184,11 +184,23 @@ VectorPtr BatchMaker::createVector<TypeKind::HUGEINT>(
     MemoryPool& pool,
     std::mt19937& gen,
     std::function<bool(vector_size_t /*index*/)> isNullAt) {
+  int bitsToMove = 0;
+  // Generate proper bits of random value for LongDecimalType tests.
+  if (type->isLongDecimal()) {
+    auto [precision, scale] = getDecimalPrecisionScale(*type);
+    int bitsNeeded = ceil(log2(std::pow(10, precision)));
+    // Round up if the bit number is not the multiples of 8 (1 byte).
+    if (bitsNeeded % 8 != 0) {
+      bitsNeeded = (bitsNeeded / 8 + 1) * 8;
```

Comment:
I think it's just 
```
int bitsToMove = 128 - ceil(log2(std::pow(10, precision)) / 8) * 8;
```

Reply:
Can I just keep it as it is if there is no major problem? I think it's more easy to understand and I already add it to another PR. Also, it looks like the `int bitsToMove = 128 - ceil(log2(std::pow(10, precision)) / 8) * 8;` is not right when Decimal(38).